src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentRenderModeLoweringPass.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Language.Components;
+
+internal class ComponentRenderModeLoweringPass : ComponentIntermediateNodePassBase, IRazorOptimizationPass
+{
+    // Run after component lowering pass
+    public override int Order => 50;
+
+    protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentIntermediateNode documentNode)
+    {
+        if (!IsComponentDocument(documentNode))
+        {
+            return;
+        }
+
+        var @namespace = documentNode.FindPrimaryNamespace();
+        var @class = documentNode.FindPrimaryClass();
+        if (@namespace == null || @class == null)
+        {
+            // Nothing to do, bail. We can't function without the standard structure.
+            return;
+        }
+
+        var references = documentNode.FindDescendantReferences<TagHelperDirectiveAttributeIntermediateNode>();
+        foreach (var reference in references)
+        {
+            if (reference is { Node: TagHelperDirectiveAttributeIntermediateNode node, Parent: ComponentIntermediateNode } && node.TagHelper.IsRenderModeTagHelper())
+            {
+                var expression = node.Children[0] switch
+                {
+                    CSharpExpressionIntermediateNode cSharpNode => cSharpNode.Children[0],
+                    IntermediateNode token => token
+                };
+
+                reference.Replace(new RenderModeIntermediateNode(expression));
```

Comment:
So I thought about using the children instead, but `RenderMode` always has exactly 1 child, so you end up having to encode that in multiple places rather than just having an explicit property for it. (Splat has a variable number of children).

Reply:
> Splat has a variable number of children

I thought it was simply an `@attributes` attribute taking a dictionary: [docs](https://learn.microsoft.com/en-us/aspnet/core/mvc/views/razor?view=aspnetcore-7.0#attributes)